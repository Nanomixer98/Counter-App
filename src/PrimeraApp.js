// import React, { Fragment } from 'react';
import React from 'react';
import PropTypes from 'prop-types';


// Functional Component (FC)
const PrimeraApp = ({ 
        saludo, 
        subtitulo 
    }) => {

    // const saludo = 123;
    // const saludo = 124.5654;
    // const saludo = false;
    // const saludo = [1,2,3,4,5,6];
    // const saludo = {
    //     nombre: "Alexis",
    //     edad: 23    
    // };
    // const saludo = "Hola Mundo desde el const";

    // console.log(props);

    // if(!saludo) {
    //     throw new Error("NO hay SALUDO")
    // }

    return (
        <>
            {/* <h1>{ JSON.stringify(saludo) }</h1> */}
            {/* <pre>{ JSON.stringify(saludo, null, 3) }</pre> */}
            <h1>{ saludo }!!!</h1>
            <p>{ subtitulo }</p>
        </>
    );
}

PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired,
};

PrimeraApp.defaultProps = {
    subtitulo: 'Esto es un subtitulo por defecto'
}

export default PrimeraApp;
