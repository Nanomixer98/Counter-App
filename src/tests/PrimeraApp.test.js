import "@testing-library/jest-dom";
import PrimeraApp from "../PrimeraApp";
import React from 'react';
import { shallow } from 'enzyme';


describe('Pruebas en <PrimeraApp/>', () => {
    
    // ! Esta prueba fue descartada porque utiliza jest puro, en su lugar usaremos Enzime
    // test('Debe de mostrar el mensaje "Hola! Soy Nanomixer ðŸ˜Ž"', () => {
      
    //     const saludo = "Hola! Soy Nanomixer ðŸ˜Ž";
    //     const { getByText } = render(<PrimeraApp saludo={saludo}/>);

    //     expect(getByText(saludo)).toBeInTheDocument();
    // });
    
    // * Esta prueba es realizada con Enzime
    test('Debe de mostrar <PrimeraApp /> correctamente', () => {
        const saludo = "Hola! Soy Nanomixer ðŸ˜Ž";
        const wrapper = shallow(<PrimeraApp saludo={ saludo }/>);

        expect(wrapper).toMatchSnapshot();
    });
    
    test('Debe de mostrar el subtitulo enviado por props', () => {
        const saludo = "Hola! Soy Nanomixer ðŸ˜Ž";
        const subTitulo = "Soy un subtitulo enviado";
        const wrapper = shallow(
            <PrimeraApp 
                saludo={ saludo }
                subtitulo={ subTitulo }
            />
        );
        const textoParrafo = wrapper.find('p').text();
        // console.log(textoParrafo);

        expect(textoParrafo).toBe(subTitulo);
    });
    

});
