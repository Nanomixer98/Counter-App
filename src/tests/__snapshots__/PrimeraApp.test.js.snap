// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PrimeraApp/> Debe de mostrar <PrimeraApp /> correctamente 1`] = `
<Fragment>
  <h1>
    Hola! Soy Nanomixer ðŸ˜Ž
    !!!
  </h1>
  <p>
    Esto es un subtitulo por defecto
  </p>
</Fragment>
`;
